### PR TITLE
Introduce Azure Container Apps

### PR DESCRIPTION
**Pull request by black-bryan**
Introduce Azure Container Apps deployment

### CODE REVIEW

File: beta/azure-container-apps/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
+# [Beta] Deploy 1Password SCIM Bridge on Azure Container Apps
+
+This deployment example describes how to deploy 1Password SCIM Bridge as an app on Azures's [Container Apps](https://azure.microsoft.com/en-us/products/container-apps/#overview) service.
+
+The deployment consists of two [containers](https://learn.microsoft.com/en-us/azure/container-apps/containers), the SCIM bridge cntainer and a Redis container and an [ingress](https://learn.microsoft.com/en-us/azure/container-apps/ingress-overview) for the SCIM bridge container.
```
Comment (plttn) [inline beta/azure-container-apps/ReadMe.md:5]: s/cntainer/container

**Review by plttn — COMMENTED** [review]
Initial review, more to come

File: beta/azure-container-apps/ReadMe.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+   - Select **Password**. 
+   - Paste the base64 value into the password field. 
+   - Rename the Password label above the value you just pasted, to something like `Base64 Value`. 
+   - Select **Save**.
+
+## Deploy 1Password SCIM Bridge to Azure Container App
+
+## Automatic Azure Deployment Steps using Azure CLI
+
+Using the Azure Cloud Shell can be easier (the assumption for the commands below is that you are using the Bash Cloud Shell), but this can also been done directly on yours system with the Azure CLI tool with the ContainerApp Extension
```
Comment (ag-adampike) [inline beta/azure-container-apps/ReadMe.md:66]: Style ([easier](https://support.1password.com/style-guide/#ease)). Can we validate the below with PowerShell and note the differences in the items where there are any?
```suggestion
We recommend using the Azure Cloud Shell, but you can also deploy using the Azure CLI tool with the ContainerApp Extension. The example commands below assume you are using Bash:
```

**Review by ag-adampike — COMMENTED** [review]
Good stuff @black-bryan! I have some ideas that I've added in, and some white space, formatting edits. I'll add some more a little later.

File: beta/azure-container-apps/ReadMe.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+### Step 1: Generate credentials for automated user provisioning with 1Password
+
+1. [Sign in](https://start.1password.com) to your account on 1Password.com.
+2. Click [Integrations](https://start.1password.com/integrations/directory) in the sidebar.
+3. Choose your identity provider from the User Provisioning section.
+4. Choose "Custom deployment".
+5. Use the "Save in 1Password" buttons for both the `scimsession` file and `bearer token` to save them as items in your 1Password account. 
+6. Also use the download icon next to the `scimsession` file to save this file on your system.
+
+### Step 2: Configure the `scimession` credentials for passing to Container App
```
Comment (scottisloud) [inline beta/azure-container-apps/ReadMe.md:43]:
```suggestion
### Step 2: Configure the `scimsession` credentials for passing to Container App
```

typo fixed

File: beta/azure-container-apps/ReadMe.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    ```
+11. Open the fqdn listed in a seperate browser tab, where you can test your connection. You should be prompted to log into the SCIM bridge with your `bearer token`.
+
+### Step 3: Follow the steps to connect your Identity provider to the SCIM bridge.
+ - [Connect your Identity Provider](https://support.1password.com/scim/#step-3-connect-your-identity-provider)
+
+
+<hr>
+
+## Manual Azure Portal Deployment Steps
```
Comment (scottisloud) [inline beta/azure-container-apps/ReadMe.md:127]: I'm wondering whether we should be front-loading this README with the portal instructions. I assume that the majority of people who will be using this deployment method skew towards preferring the GUI. Front-loading this with the portal instructions may be less intimidating.

File: beta/azure-container-apps/ReadMe.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+<details>
+<summary> Manual Azure Portal Deployment Steps</summary>
+
+### Step 1: Deploy your base SCIM bridge within Azure Container Apps
+
+1. First we create a new Container App from the [Azure Container Apps Portal](https://portal.azure.com/#view/HubsExtension/BrowseResource/resourceType/Microsoft.App%2FcontainerApps).
```
Comment (scottisloud) [inline beta/azure-container-apps/ReadMe.md:134]:
```suggestion
1. Create a new Container App from the [Azure Container Apps Portal](https://portal.azure.com/#view/HubsExtension/BrowseResource/resourceType/Microsoft.App%2FcontainerApps).
```

removing second-person language and tidying some unnecessary words.

File: beta/azure-container-apps/ReadMe.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+2. Either use an existing Resource Group or select Create New under the **Resource Group** Section.
```
Comment (scottisloud) [inline beta/azure-container-apps/ReadMe.md:135]:
```suggestion
2. Either use an existing Resource Group or select Create New under the **Resource Group** section.
```

File: beta/azure-container-apps/ReadMe.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+4. Select your **Region** for the deployment.
+5. Either use the existing Container App Environment generated or select Create New under the **Container Apps Environment** Section.
+    1. If creating a new Container App Environment, use a name that makes sense for your organization, for example op-scim-con-app-env.
+    2. Leave **Consumption** selected, and **Zone Redundancy** disabled.
+    3. Select **Create** on the create container apps environment window.
+6. Select **Next : Container >** to continue the container app creation.
+    1. Uncheck **Use quickstart image**.
+    2. Change the name of the container to *op-scim-bridge*.
+    3. Select **Docker Hub** for the image source.
+    4. Enter **1password/scim:v2.8.1** for the **Image and Tag**.
```
Comment (scottisloud) [inline beta/azure-container-apps/ReadMe.md:146]:
```suggestion
    4. Enter **1password/scim:v2.8.2** for the **Image and Tag**.
```

Time keeps on timing!

File: beta/azure-container-apps/ReadMe.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+    5. Leave the CPU and Memory as 0.25 CPU cores, 0.5 Gi memory.
```
Comment (scottisloud) [inline beta/azure-container-apps/ReadMe.md:147]: ~~This is likely fine for "standard" deployments.~~
* ~~Do we know if this scales dynamically, vertically?~~
* ~~If not, should we make it clear that this resourcing is suitable for deployments of <1000 users?~~
I see you have a Todo at the bottom for exactly this. Disregard!

File: beta/azure-container-apps/ReadMe.md [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+10. Once the deployment is complete, select **Go to Resource**.
+
+> **Note:** Expect the deployment to fail as it is expecting the second container to deploy successfully.
+
+### Step 2: Finalize your SCIM bridge deployment to use your scimsession secret and add the Redis Container.
+
+1. Within Azure Container App browser tab, Select **Secrets** along the left hand side.
+    1. Select **Add**.
+    2. Enter **scimsession** as the **Key field**.
+    3. Enter the base 64 value into the **Value** field. (This is what we added to your 1Password **scimsession file item** in the getting started section)
```
Comment (scottisloud) [inline beta/azure-container-apps/ReadMe.md:166]:
```suggestion
    3. Enter the base64 value into the **Value** field. (This is what we added to your 1Password **scimsession file item** in the getting started section)
```

File: beta/azure-container-apps/ReadMe.md [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+<summary> Manual Azure Portal Upgrade Steps</summary>
+
+1. Within Container App from the [Azure Container Apps Portal](https://portal.azure.com/#view/HubsExtension/BrowseResource/resourceType/Microsoft.App%2FcontainerApps), Select **Containers** from the left hand side.
+2. Select **Edit and deploy** along the top.
+3. Put a check in the box next to your **op-scim-bridge** container and select **Edit**.
+4. Change the version number **2.8.2** in the **Image and Tag** field, **1password/scim:v2.8.2** to match the latest version from our [SCIM Bridge Release Notes page](https://app-updates.agilebits.com/product_history/SCIM).
+5. Select **Save**.
+6. Select **Create** to deploy a new revision using the updating image.
+7. Log into your SCIM bridge URL with your bearer token to validate in the top left hand side that you are running the version of the SCIM Bridge. (logging in with the bearer token will also update your Automated User Provisioning page with the latest access time and with the current version).
+</details>
```
Comment (scottisloud) [inline beta/azure-container-apps/ReadMe.md:221]:
```suggestion
### Update 1Password SCIM Bridge

#### Update using the `az` CLI
The latest version of 1Password SCIM Bridge is posted on our [Release Notes](https://app-updates.agilebits.com/product_history/SCIM) website, where you can find details about the latest changes. 

1. Start the Azure Cloud Shell from the navigation bar of your [Azure Portal](https://portal.azure.com).
2. Run the following command, replacing $ContainerAppName ResourceGroup with the names from your deployment. Also ensure to change the version number 2.8.2 to match the latest version from our [SCIM Bridge Release Notes page](https://app-updates.agilebits.com/product_history/SCIM).
```az containerapp update -n $ContainerAppName -g $ResourceGroup --container-name op-scim-bridge --image docker.io/1password/scim:v2.8.2```
3. Log into your SCIM bridge URL with your bearer token to validate in the top left hand side that you are running the version of the SCIM Bridge. (logging in with the bearer token will also update your Automated User Provisioning page with the latest access time and with the current version).

#### Update using the Azure Portal

1. Within Container App from the [Azure Container Apps Portal](https://portal.azure.com/#view/HubsExtension/BrowseResource/resourceType/Microsoft.App%2FcontainerApps), Select **Containers** from the left hand side.
2. Select **Edit and deploy** along the top.
3. Put a check in the box next to your **op-scim-bridge** container and select **Edit**.
4. Change the version number **2.8.2** in the **Image and Tag** field, **1password/scim:v2.8.2** to match the latest version from our [SCIM Bridge Release Notes page](https://app-updates.agilebits.com/product_history/SCIM).
5. Select **Save**.
6. Select **Create** to deploy a new revision using the updating image.
7. Log into your SCIM bridge URL with your bearer token to validate in the top left hand side that you are running the version of the SCIM Bridge. (logging in with the bearer token will also update your Automated User Provisioning page with the latest access time and with the current version).
```
In line with my thinking above about emphasizing the portal/GUI method, this suggestion puts both cli and portal methods at the same level and makes the language slightly more consistent between the two sections. 


### Update 1Password SCIM Bridge

#### Update using the `az` CLI
The latest version of 1Password SCIM Bridge is posted on our [Release Notes](https://app-updates.agilebits.com/product_history/SCIM) website, where you can find details about the latest changes. 

1. Start the Azure Cloud Shell from the navigation bar of your [Azure Portal](https://portal.azure.com).
2. Run the following command, replacing $ContainerAppName ResourceGroup with the names from your deployment. Also ensure to change the version number 2.8.2 to match the latest version from our [SCIM Bridge Release Notes page](https://app-updates.agilebits.com/product_history/SCIM).
```az containerapp update -n $ContainerAppName -g $ResourceGroup --container-name op-scim-bridge --image docker.io/1password/scim:v2.8.2```
3. Log into your SCIM bridge URL with your bearer token to validate in the top left hand side that you are running the version of the SCIM Bridge. (logging in with the bearer token will also update your Automated User Provisioning page with the latest access time and with the current version).

#### Update using the Azure Portal

1. Within Container App from the [Azure Container Apps Portal](https://portal.azure.com/#view/HubsExtension/BrowseResource/resourceType/Microsoft.App%2FcontainerApps), Select **Containers** from the left hand side.
2. Select **Edit and deploy** along the top.
3. Put a check in the box next to your **op-scim-bridge** container and select **Edit**.
4. Change the version number **2.8.2** in the **Image and Tag** field, **1password/scim:v2.8.2** to match the latest version from our [SCIM Bridge Release Notes page](https://app-updates.agilebits.com/product_history/SCIM).
5. Select **Save**.
6. Select **Create** to deploy a new revision using the updating image.
7. Log into your SCIM bridge URL with your bearer token to validate in the top left hand side that you are running the version of the SCIM Bridge. (logging in with the bearer token will also update your Automated User Provisioning page with the latest access time and with the current version).

**Review by scottisloud — CHANGES_REQUESTED** [review]
This is an EXTREMELY valuable contribution to the current line up of available deployment options. We know all too well how much demand there is for lower-cost deployments with fewer dependencies and this delivers that on every front. 

Most of my comments and requested changes are either simple, or merely suggestions. Happy for you to use your discretion when it comes to the suggestions. Also happy to discuss the suggestions. Not wedded to any of them.

File: beta/azure-container-apps/ReadMe.md (hunk 1 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:5]:
```suggestion
The deployment consists of two [containers](https://learn.microsoft.com/en-us/azure/container-apps/containers), the SCIM bridge container and a Redis container and an [ingress](https://learn.microsoft.com/en-us/azure/container-apps/ingress-overview) for the SCIM bridge container.
```

File: beta/azure-container-apps/ReadMe.md [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+### Troubleshooting
+
+Logs for a container app can be viewed from the **Log Stream**, there is a separate log stream for both containers of the revision that is active. Often reviewing the logs of the **op-scim-bridge** container can help understand any startup issues.
+### Update 1Password SCIM Bridge
+
+#### Update using the `az` CLI
+The latest version of 1Password SCIM Bridge is posted on our [Release Notes](https://app-updates.agilebits.com/product_history/SCIM) website, where you can find details about the latest changes. 
+
+1. Start the Azure Cloud Shell from the navigation bar of your [Azure Portal](https://portal.azure.com).
+2. Run the following command, replacing $ContainerAppName ResourceGroup with the names from your deployment. Also ensure to change the version number 2.8.2 to match the latest version from our [SCIM Bridge Release Notes page](https://app-updates.agilebits.com/product_history/SCIM).
```
Comment (black-bryan) [inline beta/azure-container-apps/ReadMe.md:207]:
```suggestion
2. Run the following command, replacing $ContainerAppName $ResourceGroup with the names from your deployment. Also ensure to change the version number 2.8.2 to match the latest version from our [SCIM Bridge Release Notes page](https://app-updates.agilebits.com/product_history/SCIM).
```
small fix to referenced variables/values that need updating.

**Comment by ag-adampike** [comment]
Have you considered providing an [ARM or YAML template](https://learn.microsoft.com/en-us/azure/container-apps/azure-resource-manager-api-spec) for this? Would it be any more convenient?

File: beta/azure-container-apps/ReadMe.md (hunk 4 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:134]: Just to close the loop on this discussion. Since more examples in scim-examples are leading customers to command based deployments, we will prioritize this method to align with the rest of our examples. The ability to deploy with a few commands directly in the Azure Cloud Shell is a huge plus here.

File: beta/azure-container-apps/ReadMe.md (hunk 8 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:154]: Keeping this on the to-do for future improvements.

File: beta/azure-container-apps/ReadMe.md [hunk 12]
```diff
@@ -0,0 +1,233 @@
...
+    --set-env-vars 'OP_REDIS_URL=redis://localhost:6379' 'OP_SESSION=secretref:scimsession'
+    ```
+10. Find your fdqn of your deployed application, this will be known as your SCIM bridge URL:
+    ```
+    az containerapp ingress show -n $ContainerAppName -g $ResourceGroup
+    ```
+11. Open the fqdn listed in a seperate browser tab, where you can test your connection. You should be prompted to log into the SCIM bridge with your `bearer token`.
+
+### Step 3: Follow the steps to connect your Identity provider to the SCIM bridge.
+ - [Connect your Identity Provider](https://support.1password.com/scim/#step-3-connect-your-identity-provider)
```
Comment (black-bryan) [inline beta/azure-container-apps/ReadMe.md:129]: Trying to commit Pike suggested changes

```suggestion
1. Start the Azure Cloud Shell from the navigation bar of your [Azure Portal](https://portal.azure.com).
2. Define variables for the deployment using the following example in the Cloud Shell. Update the values in a text editor before pasting it into the terminal:

   ```bash
   ResourceGroup="op-scim-bridge-rg"
   Location="canadacentral"
   ContainerAppEnvironment="op-scim-bridge-con-app-env"
   ContainerAppName="op-scim-bridge-con-app"
   ```

3. Create the resource group:

   ```bash
   az group create --name $ResourceGroup --location $Location
   ```

4. Create the Container App Environment:

   ```bash
   az containerapp env create --name $ContainerAppEnvironment --resource-group $ResourceGroup --location $Location
   ```

5. Upload your `scimsession` file to the Cloud Shell:
   - Click the “Upload/Download files” button and choose Upload.
   - Find the `scimsession` file that you saved to your computer and choose it.
   - Make a note of the upload destination, then click Complete.

6. Create the base Container App and Secret for your Base64-encoded `scimsession` credentials:

   ```bash
    az containerapp create -n $ContainerAppName -g $ResourceGroup \
        --container-name op-scim-bridge \
        --image docker.io/1password/scim:v2.8.2 \
        --environment $ContainerAppEnvironment \
        --ingress external --target-port 3002 \
        --cpu 0.25 --memory 0.5Gi \
        --min-replicas 1 --max-replicas 1 \
        --secrets scimsession="$(cat /home/$USER/scimsession | base64)" \
        --env-vars OP_REDIS_URL="redis://localhost:6379" OP_SESSION=secretref:scimsession
    ```

    > **Note**
    >
    > You may be prompted to install the ContainerApp extension with this message:
    >
    > ```bash
    > The command requires the extension containerapp. Do you want to install it now? The command will continue to run after the extension is installed. (Y/n):
    > ```
    >
    > The ContainerApp extension is required to deploy Container Apps from the command line. Type `Y` and press enter to install the extension.

7. Update your Container App to add the Redis container and get the fully qualified domain name to use as the URL for your SCIM bridge:

   ```bash
   az containerapp update -n $ContainerAppName -g $ResourceGroup \
       --container-name op-scim-redis \
       --image docker.io/redis \
       --cpu 0.25 --memory 0.5Gi \
       --set-env-vars REDIS_ARGS="--maxmemory 256mb --maxmemory-policy volatile-lru" \
       --query properties.configuration.ingress.fqdn
   ```

8. Open the domain name listed in a separate browser tab to test your connection. You can sign in to your SCIM bridge using your bearer token.
```

File: beta/azure-container-apps/ReadMe.md [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+    ```
+9. Update your Container App to allow your SCIM bridge container to use the secret you have created:
+    ```
+    az containerapp update -n $ContainerAppName -g $ResourceGroup --container-name op-scim-bridge \
+    --set-env-vars 'OP_REDIS_URL=redis://localhost:6379' 'OP_SESSION=secretref:scimsession'
+    ```
+10. Find your fdqn of your deployed application, this will be known as your SCIM bridge URL:
+    ```
+    az containerapp ingress show -n $ContainerAppName -g $ResourceGroup
+    ```
```
[the remark this answers is not in the document]
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md]: Suggestions not directly being applied, I am bring your suggested changes in a separate commit.

File: beta/azure-container-apps/ReadMe.md (hunk 2 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:50]: Updated to provide Powershell options for all steps for the Azure Cloud Shell. 
```suggestion
Using the Azure Cloud Shell can be easier, but this can also been done directly on yours system with the Azure CLI tool with the ContainerApp Extension
```

File: beta/azure-container-apps/ReadMe.md (hunk 4 above)
Comment (scottisloud) [reply beta/azure-container-apps/ReadMe.md:152]: Agreed in light of our offline discussions.

File: beta/azure-container-apps/ReadMe.md [hunk 14]
```diff
@@ -0,0 +1,279 @@
...
+
+## Overview
+
+Deploying 1Password SCIM Bridge on Azure Container Apps comes with a few benefits:
+
+- For standard deployments, Azure Container Apps will host your SCIM bridge for ~$16 USD/month (when this document was written)
+  > **Note**
+  >
+  > Conatiner Apps pricing is variable based on activity. Details can be found on [Microsoft's pricing page](https://azure.microsoft.com/en-us/pricing/details/container-apps/).
+- No DNS management, and an obsecure DNS record. Azure Container Apps automatically provides a unique URL for your SCIM bridge.
```
Comment (plttn) [inline beta/azure-container-apps/ReadMe.md:19]: I'd rephrase this slightly to make it more clear.

> No DNS records required. Azure Container Apps automatically provides a unique URL for your SCIM bridge.

File: beta/azure-container-apps/ReadMe.md [hunk 15]
```diff
@@ -0,0 +1,279 @@
...
+- [`README.md`](./README.md): the document that you are reading.
+
+## Overview
+
+Deploying 1Password SCIM Bridge on Azure Container Apps comes with a few benefits:
+
+- For standard deployments, Azure Container Apps will host your SCIM bridge for ~$16 USD/month (when this document was written)
+  > **Note**
+  >
+  > Conatiner Apps pricing is variable based on activity. Details can be found on [Microsoft's pricing page](https://azure.microsoft.com/en-us/pricing/details/container-apps/).
```
Comment (plttn) [inline beta/azure-container-apps/ReadMe.md:18]: Container*

File: beta/azure-container-apps/ReadMe.md [hunk 16]
```diff
@@ -0,0 +1,279 @@
...
+## Getting started
+
+### Step 1: Generate credentials for automated user provisioning with 1Password
+
+1. [Sign in](https://start.1password.com) to your account on 1Password.com.
+2. Click [Integrations](https://start.1password.com/integrations/directory) in the sidebar.
+3. Choose your identity provider from the User Provisioning section.
+4. Choose "Custom deployment".
+5. Use the "Save in 1Password" buttons for both the `scimsession` file and `bearer token` to save them as items in your 1Password account. 
+6. Also use the download icon next to the `scimsession` file to save this file on your system.
```
Comment (plttn) [inline beta/azure-container-apps/ReadMe.md:44]: Drop `also`. It's a step, so it's clear that it needs to be done

**Review by scottisloud — DISMISSED** [review]
With the understanding that there are some lingering Todos (namely vertically scaling and Workspace) that can be addressed later, I'm happy to approve.

File: beta/azure-container-apps/ReadMe.md [hunk 17]
```diff
@@ -0,0 +1,279 @@
...
+   ```
+   az group create --name $ResourceGroup --location $Location
+   ```
+
+4. Create the Container App Environment:
+
+   ```
+   az containerapp env create --name $ContainerAppEnvironment --resource-group $ResourceGroup --location $Location
+   ```
+
```
Comment (plttn) [inline beta/azure-container-apps/ReadMe.md:82]: Should specify that this is for PowerShell.

File: beta/azure-container-apps/ReadMe.md [hunk 18]
```diff
@@ -0,0 +1,279 @@
...
+4. Select your **Region** for the deployment.
+5. Either use the existing Container App Environment generated or select Create New under the **Container Apps Environment** Section.
+    1. If creating a new Container App Environment, use a name that makes sense for your organization, for example op-scim-con-app-env.
+    2. Leave **Consumption** selected, and **Zone Redundancy** disabled.
+    3. Select **Create** on the create container apps environment window.
+6. Select **Next : Container >** to continue the container app creation.
+    1. Uncheck **Use quickstart image**.
+    2. Change the name of the container to *op-scim-bridge*.
+    3. Select **Docker Hub** for the image source.
+    4. Enter **1password/scim:v2.8.2** for the **Image and Tag**.
```
Comment (plttn) [inline beta/azure-container-apps/ReadMe.md:199]: Use `1password/scim:v2.8.2` rather than bold/italics.

File: beta/azure-container-apps/ReadMe.md [hunk 19]
```diff
@@ -0,0 +1,279 @@
...
+    5. Leave the CPU and Memory as 0.25 CPU cores, 0.5 Gi memory.
+    6. Add the following Environment variables:
+        1. **OP_SESSION** currently with of **""** for now.
+        2. **OP_REDIS_URL** with a value of **redis://localhost:6379**
```
Comment (plttn) [inline beta/azure-container-apps/ReadMe.md:203]: Ditto as above, things that should be entered as is use backticks

File: beta/azure-container-apps/ReadMe.md [hunk 20]
```diff
@@ -0,0 +1,279 @@
...
+    2. Put a check in the box next to your **op-scim-bridge** container and select **Edit**.
+    3. Scroll down the the Environment variables and edit the **OP_SESSION** variable to have a **Source** of **Reference a secret** and for the **Value**, select the **scimsession** secret.
+    4. Select **Save**.
+    5. Select the **+ Add** button, selecting **App container**.
+    6. Change the name of the container to **op-scim-redis**.
+    7. Select **Docker Hub** for the image source.
+    8. Enter **redis** for the **Image and Tag**.
+    9. Change the CPU and Memory to 0.25 CPU cores, 0.5 Gi memory.
+    10. Add the following Environment variables:
+        1. **REDIS_ARGS** with of **--maxmemory 256mb --maxmemory-policy volatile-lru**.
```
Comment (plttn) [inline beta/azure-container-apps/ReadMe.md:233]: ditto

**Review by plttn — CHANGES_REQUESTED** [review]
relatively minor changes required

File: beta/azure-container-apps/ReadMe.md (hunk 14 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:19]: Perhaps a slightly different approach based on your suggestion. 
```suggestion
- No manual DNS record creation required. Azure Container Apps automatically provides a unique DNS record for your SCIM bridge URL.
```

File: beta/azure-container-apps/ReadMe.md (hunk 15 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:18]: Fixing spelling based on @plttn's suggestion. 
```suggestion
  > Container Apps pricing is variable based on activity. Details can be found on [Microsoft's pricing page](https://azure.microsoft.com/en-us/pricing/details/container-apps/).
```

File: beta/azure-container-apps/ReadMe.md (hunk 16 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:44]: Applying @plttn suggestion.

```suggestion
6. Use the download icon next to the `scimsession` file to save this file on your system.
```

File: beta/azure-container-apps/ReadMe.md (hunk 17 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:82]: These two commands actually work for both PowerShell and bash, so no need to have separate code boxes.

File: beta/azure-container-apps/ReadMe.md (hunk 18 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:199]: Applying @plttn's suggestion.

```suggestion
    4. Enter `1password/scim:v2.8.2` for the **Image and Tag**.
```

File: beta/azure-container-apps/ReadMe.md (hunk 19 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:203]: Applying formatting suggestion

```suggestion
        1. `OP_SESSION` currently with of `""` for now.
        2. `OP_REDIS_URL` with a value of `redis://localhost:6379`
```

File: beta/azure-container-apps/ReadMe.md (hunk 20 above)
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:233]: Applying formatting suggestion

```suggestion
        1. `REDIS_ARGS` with of `--maxmemory 256mb --maxmemory-policy volatile-lru`.
```

File: beta/azure-container-apps/ReadMe.md [hunk 21]
```diff
@@ -0,0 +1,279 @@
...
+    2. Put a check in the box next to your **op-scim-bridge** container and select **Edit**.
+    3. Scroll down the the Environment variables and edit the **OP_SESSION** variable to have a **Source** of **Reference a secret** and for the **Value**, select the **scimsession** secret.
+    4. Select **Save**.
+    5. Select the **+ Add** button, selecting **App container**.
+    6. Change the name of the container to **op-scim-redis**.
+    7. Select **Docker Hub** for the image source.
+    8. Enter **redis** for the **Image and Tag**.
+    9. Change the CPU and Memory to 0.25 CPU cores, 0.5 Gi memory.
+    10. Add the following Environment variables:
+        1. `REDIS_ARGS` with of `--maxmemory 256mb --maxmemory-policy volatile-lru`.
```
Comment (black-bryan) [inline beta/azure-container-apps/ReadMe.md:233]:
```suggestion
        1. `REDIS_ARGS` with value of `--maxmemory 256mb --maxmemory-policy volatile-lru`.
```

File: beta/azure-container-apps/ReadMe.md [hunk 22]
```diff
@@ -0,0 +1,279 @@
...
+    1. Select **Edit and deploy** along the top.
+    2. Put a check in the box next to your **op-scim-bridge** container and select **Edit**.
+    3. Scroll down the the Environment variables and edit the **OP_SESSION** variable to have a **Source** of **Reference a secret** and for the **Value**, select the **scimsession** secret.
+    4. Select **Save**.
+    5. Select the **+ Add** button, selecting **App container**.
+    6. Change the name of the container to **op-scim-redis**.
+    7. Select **Docker Hub** for the image source.
+    8. Enter **redis** for the **Image and Tag**.
+    9. Change the CPU and Memory to 0.25 CPU cores, 0.5 Gi memory.
+    10. Add the following Environment variables:
```
Comment (black-bryan) [inline beta/azure-container-apps/ReadMe.md:232]:
```suggestion
    10. Add the following **Environment variables**:
```

**Review by plttn — CHANGES_REQUESTED** [review]

File: beta/azure-container-apps/ReadMe.md [hunk 23]
```diff
@@ -0,0 +1,279 @@
...
+<hr>
+
+## Manual Azure Portal Deployment Steps
+
+<details>
+<summary> Manual Azure Portal Deployment Steps</summary>
+
+### Step 1: Configure the `scimsession` credentials for passing to Container App
+
+The `scimsession` credentials will be saved as an secret variable in Container App. These credentials have to be Base64-encoded to pass them into the environment, but they're saved as a plain-text file when you download them or save in 1Password during the setup.
```
[the remark this answers is not in the document]
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:159]: applying @plttn suggestion
```suggestion
The `scimsession` credentials will be saved as a secret variable in Container App. These credentials have to be Base64-encoded to pass them into the environment, but they're saved as a plain-text file when you download them or save in 1Password during the setup.
```

File: beta/azure-container-apps/ReadMe.md [hunk 24]
```diff
@@ -0,0 +1,279 @@
...
+1. [Sign in](https://start.1password.com) to your account on 1Password.com.
+2. Click [Integrations](https://start.1password.com/integrations/directory) in the sidebar.
+3. Choose your identity provider from the User Provisioning section.
+4. Choose "Custom deployment".
+5. Use the "Save in 1Password" buttons for both the `scimsession` file and `bearer token` to save them as items in your 1Password account. 
+6. Use the download icon next to the `scimsession` file to save this file on your system.
+
+## Deploy 1Password SCIM Bridge to Azure Container App
+
+## Automatic Azure Deployment Steps using Azure CLI
```
[the remark this answers is not in the document]
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:48]: Applying @plttn suggestion
```suggestion
## Deploy 1Password SCIM Bridge to Azure Container App

### Automatic Azure Deployment Steps using Azure CLI
```

**Review by plttn — APPROVED** [review]
LGTM pending any potential technical writing touches to be done.
